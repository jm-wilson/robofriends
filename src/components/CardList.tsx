import React from 'react';
import Card from './Card';
import { IRobot } from '../containers/App';

interface ICardListProps {
  robots: IRobot[];
}

const CardList: React.FC<ICardListProps> = ({ robots }): JSX.Element => {
  return !robots.length ? (
    <div></div>
  ) : (
    <div>
      {robots.map((user, i) => {
        return <Card key={i} id={robots[i].id} name={robots[i].name} email={robots[i].email} />;
      })}
    </div>
  );
};

export default CardList;
