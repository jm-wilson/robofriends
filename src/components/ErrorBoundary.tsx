import React, { Component, ReactNode } from 'react';

interface IErrorBoundaryState {
  hasError: boolean;
}

interface IErrorBoundaryProps {
  children: ReactNode;
}

class ErrorBoundry extends Component {
  state: IErrorBoundaryState;

  constructor(props: IErrorBoundaryProps) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return <h1>Oooops. That is not good</h1>;
    }
    return this.props.children;
  }
}

export default ErrorBoundry;
