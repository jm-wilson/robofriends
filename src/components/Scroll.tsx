import React from 'react';

interface IScrollProps {
  children?: JSX.Element;
}
const Scroll: React.FC<IScrollProps> = (props): JSX.Element => {
  return (
    <div style={{ overflowY: 'scroll', border: '3px solid black', height: '800px' }}>
      {props.children}
    </div>
  );
};

export default Scroll;
