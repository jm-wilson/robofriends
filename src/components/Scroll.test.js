import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';

it('should render Scroll component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Snow',
      email: 'john@gmail.com',
    },
  ];
  expect(
    shallow(
      <Scroll>
        <ErrorBoundary>
          <CardList robots={mockRobots} />
        </ErrorBoundary>
      </Scroll>,
    ),
  ).toMatchSnapshot();
});
