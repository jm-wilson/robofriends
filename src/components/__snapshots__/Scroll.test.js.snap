// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Scroll component 1`] = `
<div
  style={
    Object {
      "border": "3px solid black",
      "height": "800px",
      "overflowY": "scroll",
    }
  }
>
  <ErrorBoundry>
    <CardList
      robots={
        Array [
          Object {
            "email": "john@gmail.com",
            "id": 1,
            "name": "John Snow",
          },
        ]
      }
    />
  </ErrorBoundry>
</div>
`;
